### ISSUE TITLE
Delete vs Exclude (or whatever)

### ISSUE THREAD

**Issue by karlic** [kolyvan/kybook#61]
I've opened this as a separate issue as it is quite important and actually different from what I meant in #57.

> Yes, some inconsistency in terms is exist.
> 
> - Delete empty action in Cataloguer screen
> - Delete menu item in Cataloguer screen by tapping on a category
> - Delete a category in BookCard during editing
> - Exclude menu item in Cataloguer screen by tapping on a category
> - Something I forgot
> 
> I do not like Exclude, it's the compromise because of existing (or better to say: must have a distinct between) two different scenarios: delete books with some category and delete only a category itself.
> Ideally, delete verb must be used only for a destructive action (when books will be deleted).
> But in the real-life users do not understand a term Exclude
> Using Exclude (or any other similar term) where it must be used logically will just confuse an ordinal user and we must use Delete term despite of illogical.

**Comment by karlic** [comment]
I agree, you have to differentiate between deleting books and removing books from a category. Also 'Exclude' is not ideal. But put that to one side for now.

> Delete empty action in Cataloguer screen

You could get rid of this menu item if the app **_always_** cleaned up empty categories. (It does it #57, but not in #40)
OR
you could make it clearer what it does by having the word 'Empty' rather than the (#) icon (hash in a circle) in empty categories

'Delete' is actually correct here, because you are deleting empty category names from the device.

**Comment by karlic** [comment]
> Delete a category in BookCard during editing

The word 'delete' is wrong here as you are actually excluding the book from the category. This should say 'Exclude' until you've decided what word to use.

**Comment by karlic** [comment]
> Delete menu item in Cataloguer screen by **_long_** tapping on a category
> Exclude menu item in Cataloguer screen by **_long_** tapping on a category

If I long tap on Delete I get a dialogue 'Delete bookshelf? {bookshelf_name} Cancel Delete', it is **_not_** just deleting a bookshelf/category it is deleting books from the device. It should say, 'Delete x book(s) from the device? This cannot be undone. Cancel Delete' (You don't need to mention the bookshelf/category if the app tidies up.) 

If I long tap on Exclude there is no dialogue box. This should say, 'Exclude x book(s) from the bookshelf/category {bookshelf_name}? Cancel Exclude'

ALTERNATIVELY, you could have a one-time pop-up, the first time the user long taps a category that says, 'Exclude excludes the selected books from the category. Delete deletes the selected books from the device (and cannot be undone). OK'

**Comment by karlic** [comment]
I agree 'Delete' sounds destructive and makes me think I will lose data (books), so Delete is correct for deleting books from the device. The trash can icon reinforces that this is destructive.

Don't think of deleting categories, think of taking books out of a category and empty categories get cleaned up. (Like in programming, you don't delete memory, you free it up and the garbage collector cleans up after you.)
In which case, Exclude needs to mean 'take out' or the opposite of 'put in':
unassign, uncategorize, unallocate, etc. www.thesaurus.com

Alternatively, try and think about real books. In which case taking a book out of a category is taking it off a particular shelf. Deleting books is putting them in the rubbish bin, so:
'Exclude' becomes 'Unshelve'
'Delete' becomes 'Trash' (actually 'Bin' in UK English, but 'Trash' is more common in computing scenarios). (Users will be familiar with trash cans on Windows, Macs, or Linux, so they'll know that means the book(s) will be lost.)

**Comment by kolyvan** [comment]
Thank you very much for suggestions.

**Comment by karlic** [comment]
Another thought.
'Unfile' instead of 'Exclude', especially if you changed 'Add a subject/author/etc.' to 'File under [another] subject/author/etc.' in the book card. You could probably find/make a nice icon with a bookshelf with a book being taken out.

**Comment by karlic** [comment]
More thoughts.
'Cataloguer' could be renamed 'Shelves' (or 'Bookshelves') (it's on the Library screen, so it's logical because that would be the next level down in a real library). 
'Exclude' then becomes 'Unshelve' and (in the book card) 'Add an author' becomes, 'Shelve under author' (likewise with subject, etc.). That would follow the real world quite well. 

By the way, I'm not entirely convinced of the need for 'Cataloguer' **_and_** 'iPad' **_and_** the Folders you can add to the Library screen. Why can't the Cataloguer ('Shelves') be the only icon there and 'iPad' and other Folders be under Cataloguer as a catalogue ('Shelf') called 'Locations'.

Personally, because I'm often without Wi-Fi the Folders are of little (no?) use to me and I just copy my books on to the iPad

Personally, I very much like the layout of the 'iPad' view, so if you **_did_** combine everything under Cataloguer, I'd like to have the option for it to open with the 'All books' view by default. :-)

**Comment by kolyvan** [comment]
> 'Cataloguer' could be renamed 'Shelves'

No, Cataloguer contains not only shelves but also list of categories like authors, etc.

>  'Cataloguer' and 'iPad' 

Because, `Cataloguer` is an interface to the Database de facto, and `iPad` or `iCloud` or `Google Drive` is a interface to physical file storage. There are different beasts.

**Comment by karlic** [comment]
> No, Cataloguer contains not only shelves but also list of categories like authors, etc.

Tap Catalogue -> Authors -> Konstantin Boukreev and you are looking at a **_shelf_** of books by Konstantin.
In the UK, books can be shelved by authors, bestsellers, returns, new, size, even colour! (http://www.tenpennydreams.com/2014/10/18/rainbow-rooms-tips-colour-coding-bookshelves/)

> Because, Cataloguer is an interface to the Database de facto, and iPad or iCloud or Google Drive is a interface to physical file storage. There are different beasts.

Yes—to a programmer. Would your granny or 10-year-old understand ‘interface to physical file storage’ or even care when they are using the app to read a book? Copying the real world makes it easier for users.
